executor/aggregate.go

```diff
@@ -1079,15 +1079,16 @@ func (e *vecGroupChecker) splitIntoGroups(chk *chunk.Chunk) (areSameGroup bool,
 	if len(e.previousLastGroupKey) == 0 {
 		areSameGroup = false
 	} else {
-		groupKey0 := e.previousLastGroupKey
-		groupKey1 := string(hack.String(e.firstGroupKey))
-		if groupKey0 == groupKey1 {
+		if bytes.Equal(e.previousLastGroupKey, e.firstGroupKey) {
 			areSameGroup = true
 		} else {
 			areSameGroup = false
 		}
 	}
-	e.previousLastGroupKey = string(e.lastGroupKey)
+	if cap(e.previousLastGroupKey) < len(e.lastGroupKey) {
```

Comment:
If the length of `e.previousLastGroupKey` is less than length of `e.lastGroupKey`, it will copy fewer bytes into e.previousLastGroupKey. See https://play.golang.org/p/HRzdKtFRg6V

suggestion
```go
if length := len(e.previousLastGroupKey); length >= len(e.lastGroupKey) {
    e.previousLastGroupKey = e.previousLastGroupKey[:length]
} else {
    e.previousLastGroupKey = make([]byte, len(e.lastGroupKey))
}